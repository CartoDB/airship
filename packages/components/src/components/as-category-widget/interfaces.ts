export interface Category {
  name: string;
  value: number;
  color?: string;
}
export interface CategoryOptions {
  maximumValue: number;
  isOther?: boolean;
}
