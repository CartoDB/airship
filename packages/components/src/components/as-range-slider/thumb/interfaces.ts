export interface Thumb {
  value: number;
  valueMin: number;
  valueMax: number;
  percentage: number;
}
