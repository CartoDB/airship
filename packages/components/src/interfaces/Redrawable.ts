export interface Redrawable {
  redraw(): void;
}
