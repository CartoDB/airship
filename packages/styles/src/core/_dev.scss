@import 'tools/_mixins';
@import 'variables/_variables';
