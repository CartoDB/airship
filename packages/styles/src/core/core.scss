@import 'variables/_variables';

* {
  box-sizing: border-box;
  font-family: sans-serif;
}

:root {
  --primary-color: #{$primary-color};
}
