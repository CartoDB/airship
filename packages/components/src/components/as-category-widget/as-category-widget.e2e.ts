import { E2EElement, E2EPage, newE2EPage } from '@stencil/core/testing';
import { Category } from './interfaces';

describe('as-category-widget', () => {
  let page: E2EPage;

  beforeEach(async () => {
    page = await newE2EPage();

    await page.setContent(`
      <as-category-widget></as-category-widget>
    `);
  });

  describe('Rendering', async () => {
    it('should render properly', async () => {
      const element: E2EElement = await page.find('as-category-widget');
      element.setProperty('heading', 'Category Widget Example');
      element.setProperty('description', 'Description for Category Widget');
      element.setProperty('categories', exampleCategories);
      element.setProperty('showClearButton', true);

      await page.waitForChanges();

      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should not render header when showHeader is false', async () => {
      const element: E2EElement = await page.find('as-category-widget');
      element.setProperty('heading', 'Category Widget Example');
      element.setProperty('description', 'Description for Category Widget');
      element.setProperty('showHeader', false);

      await page.waitForChanges();

      const actual = await page.find('.as-histogram-widget__header');

      expect(actual).toBeFalsy();
    });

    it('should render clear button', async () => {
      const element: E2EElement = await page.find('as-category-widget');
      element.setProperty('showClearButton', true);
      element.setProperty('categories', exampleCategories);

      await page.waitForChanges();

      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should render bars according to total value', async () => {
      const element: E2EElement = await page.find('as-category-widget');
      element.setProperty('categories', exampleCategories);
      element.setProperty('useTotalPercentage', true);

      await page.waitForChanges();

      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should format display value when formatValue prop is present', async () => {
      const element: E2EElement = await page.find('as-category-widget');
      element.setProperty('categories', exampleCategories);
      element.setProperty('valueFormatter', (value) => `${value}â‚¬`);

      await page.waitForChanges();

      expect(element.outerHTML).toMatchSnapshot();
    });
  });

  describe('Behaviour', async () => {
    it('should emit an event containing selected categories when a new category is selected', async () => {
      const element: E2EElement = await page.find('as-category-widget');
      element.setProperty('categories', exampleCategories);

      await page.waitForChanges();

      const categoriesSelectedSpy = await page.spyOnEvent('categoriesSelected');

      await page.click('.as-category-widget__category');
      await page.waitForChanges();

      expect(categoriesSelectedSpy).toHaveReceivedEventDetail([exampleCategories[0].name]);
    });

    it('should emit an event containing selected categories when categories are cleared', async () => {
      const element: E2EElement = await page.find('as-category-widget');
      element.setProperty('categories', exampleCategories);
      element.setProperty('showClearButton', true);

      await page.waitForChanges();

      await page.click('.as-category-widget__category');
      await page.waitForChanges();

      const categoriesSelectedSpy = await page.spyOnEvent('categoriesSelected');

      await page.click('.as-category-widget__clear');
      await page.waitForChanges();

      expect(categoriesSelectedSpy).toHaveReceivedEventDetail([]);
    });
  });
});

const exampleCategories: Category[] = [
  { name: 'Bars & Restaurants', value: 1000, color: '#FABADA' },
  { name: 'Fashion', value: 900 },
  { name: 'Grocery', value: 800 },
  { name: 'Health', value: 400 },
  { name: 'Shopping mall', value: 250 },
  { name: 'Transportation', value: 1550 },
  { name: 'Leisure', value: 1200 }
];
