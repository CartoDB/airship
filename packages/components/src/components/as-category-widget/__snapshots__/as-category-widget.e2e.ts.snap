// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`as-category-widget Rendering should format display value when formatValue prop is present 1`] = `"<as-category-widget class=\\"hydrated\\"><as-widget-header is-empty=\\"\\" no-data-message=\\"NO DATA AVAILABLE\\" class=\\"hydrated\\"><h2 class=\\"as-widget-header__header\\"></h2><p class=\\"as-widget-header__subheader as-widget-header__subheader--empty as-body\\">NO DATA AVAILABLE</p></as-widget-header><p class=\\"content as-body\\">There is no data to display.</p></as-category-widget>"`;

exports[`as-category-widget Rendering should render bars according to total value 1`] = `"<as-category-widget class=\\"hydrated\\"><as-widget-header no-data-message=\\"NO DATA AVAILABLE\\" class=\\"hydrated\\"><h2 class=\\"as-widget-header__header\\"></h2><p class=\\"as-widget-header__subheader as-body\\"></p></as-widget-header><ul class=\\"as-category-widget__list\\"><li class=\\"as-category-widget__category\\"><p class=\\"as-category-widget__info as-body\\"></p><div class=\\"as-category-widget__title\\">Bars &amp; Restaurants</div> 1.0K<p></p><div class=\\"as-category-widget__bar\\"><div class=\\"as-category-widget__bar-value\\" style=\\"background-color: rgb(250, 186, 218); width: 16.3934%;\\"></div></div></li><li class=\\"as-category-widget__category\\"><p class=\\"as-category-widget__info as-body\\"></p><div class=\\"as-category-widget__title\\">Fashion</div>   900<p></p><div class=\\"as-category-widget__bar\\"><div class=\\"as-category-widget__bar-value\\" style=\\"background-color: var(--category-bar--color); width: 14.7541%;\\"></div></div></li><li class=\\"as-category-widget__category\\"><p class=\\"as-category-widget__info as-body\\"></p><div class=\\"as-category-widget__title\\">Grocery</div>   800<p></p><div class=\\"as-category-widget__bar\\"><div class=\\"as-category-widget__bar-value\\" style=\\"background-color: var(--category-bar--color); width: 13.1148%;\\"></div></div></li><li class=\\"as-category-widget__category\\"><p class=\\"as-category-widget__info as-body\\"></p><div class=\\"as-category-widget__title\\">Health</div>   400<p></p><div class=\\"as-category-widget__bar\\"><div class=\\"as-category-widget__bar-value\\" style=\\"background-color: var(--category-bar--color); width: 6.55738%;\\"></div></div></li><li class=\\"as-category-widget__category\\"><p class=\\"as-category-widget__info as-body\\"></p><div class=\\"as-category-widget__title\\">Shopping mall</div>   250<p></p><div class=\\"as-category-widget__bar\\"><div class=\\"as-category-widget__bar-value\\" style=\\"background-color: var(--category-bar--color); width: 4.09836%;\\"></div></div></li><li class=\\"as-category-widget__category\\"><p class=\\"as-category-widget__info as-body\\"></p><div class=\\"as-category-widget__title\\">Transportation</div> 1.6K<p></p><div class=\\"as-category-widget__bar\\"><div class=\\"as-category-widget__bar-value\\" style=\\"background-color: var(--category-bar--color); width: 25.4098%;\\"></div></div></li><li class=\\"as-category-widget__category\\"><p class=\\"as-category-widget__info as-body\\"></p><div class=\\"as-category-widget__title\\">Leisure</div> 1.2K<p></p><div class=\\"as-category-widget__bar\\"><div class=\\"as-category-widget__bar-value\\" style=\\"background-color: var(--category-bar--color); width: 19.6721%;\\"></div></div></li></ul><footer class=\\"as-category-widget__footer\\"><div class=\\"as-category-widget__count as-body\\">All selected</div></footer></as-category-widget>"`;

exports[`as-category-widget Rendering should render clear button 1`] = `"<as-category-widget class=\\"hydrated\\"><as-widget-header no-data-message=\\"NO DATA AVAILABLE\\" class=\\"hydrated\\"><h2 class=\\"as-widget-header__header\\"></h2><p class=\\"as-widget-header__subheader as-body\\"></p></as-widget-header><ul class=\\"as-category-widget__list\\"><li class=\\"as-category-widget__category\\"><p class=\\"as-category-widget__info as-body\\"></p><div class=\\"as-category-widget__title\\">Bars &amp; Restaurants</div> 1.0K<p></p><div class=\\"as-category-widget__bar\\"><div class=\\"as-category-widget__bar-value\\" style=\\"background-color: rgb(250, 186, 218); width: 64.5161%;\\"></div></div></li><li class=\\"as-category-widget__category\\"><p class=\\"as-category-widget__info as-body\\"></p><div class=\\"as-category-widget__title\\">Fashion</div>   900<p></p><div class=\\"as-category-widget__bar\\"><div class=\\"as-category-widget__bar-value\\" style=\\"background-color: var(--category-bar--color); width: 58.0645%;\\"></div></div></li><li class=\\"as-category-widget__category\\"><p class=\\"as-category-widget__info as-body\\"></p><div class=\\"as-category-widget__title\\">Grocery</div>   800<p></p><div class=\\"as-category-widget__bar\\"><div class=\\"as-category-widget__bar-value\\" style=\\"background-color: var(--category-bar--color); width: 51.6129%;\\"></div></div></li><li class=\\"as-category-widget__category\\"><p class=\\"as-category-widget__info as-body\\"></p><div class=\\"as-category-widget__title\\">Health</div>   400<p></p><div class=\\"as-category-widget__bar\\"><div class=\\"as-category-widget__bar-value\\" style=\\"background-color: var(--category-bar--color); width: 25.8065%;\\"></div></div></li><li class=\\"as-category-widget__category\\"><p class=\\"as-category-widget__info as-body\\"></p><div class=\\"as-category-widget__title\\">Shopping mall</div>   250<p></p><div class=\\"as-category-widget__bar\\"><div class=\\"as-category-widget__bar-value\\" style=\\"background-color: var(--category-bar--color); width: 16.129%;\\"></div></div></li><li class=\\"as-category-widget__category\\"><p class=\\"as-category-widget__info as-body\\"></p><div class=\\"as-category-widget__title\\">Transportation</div> 1.6K<p></p><div class=\\"as-category-widget__bar\\"><div class=\\"as-category-widget__bar-value\\" style=\\"background-color: var(--category-bar--color); width: 100%;\\"></div></div></li><li class=\\"as-category-widget__category\\"><p class=\\"as-category-widget__info as-body\\"></p><div class=\\"as-category-widget__title\\">Leisure</div> 1.2K<p></p><div class=\\"as-category-widget__bar\\"><div class=\\"as-category-widget__bar-value\\" style=\\"background-color: var(--category-bar--color); width: 77.4194%;\\"></div></div></li></ul><footer class=\\"as-category-widget__footer\\"><div class=\\"as-category-widget__count as-body\\">All selected</div><button class=\\"as-btn as-btn--primary as-btn--s as-category-widget__clear\\" disabled=\\"\\">Clear selection</button></footer></as-category-widget>"`;

exports[`as-category-widget Rendering should render properly 1`] = `"<as-category-widget class=\\"hydrated\\"><as-widget-header no-data-message=\\"NO DATA AVAILABLE\\" class=\\"hydrated\\"><h2 class=\\"as-widget-header__header\\">Category Widget Example</h2><p class=\\"as-widget-header__subheader as-body\\">Description for Category Widget</p></as-widget-header><ul class=\\"as-category-widget__list\\"><li class=\\"as-category-widget__category\\"><p class=\\"as-category-widget__info as-body\\"></p><div class=\\"as-category-widget__title\\">Bars &amp; Restaurants</div> 1.0K<p></p><div class=\\"as-category-widget__bar\\"><div class=\\"as-category-widget__bar-value\\" style=\\"background-color: rgb(250, 186, 218); width: 64.5161%;\\"></div></div></li><li class=\\"as-category-widget__category\\"><p class=\\"as-category-widget__info as-body\\"></p><div class=\\"as-category-widget__title\\">Fashion</div>   900<p></p><div class=\\"as-category-widget__bar\\"><div class=\\"as-category-widget__bar-value\\" style=\\"background-color: var(--category-bar--color); width: 58.0645%;\\"></div></div></li><li class=\\"as-category-widget__category\\"><p class=\\"as-category-widget__info as-body\\"></p><div class=\\"as-category-widget__title\\">Grocery</div>   800<p></p><div class=\\"as-category-widget__bar\\"><div class=\\"as-category-widget__bar-value\\" style=\\"background-color: var(--category-bar--color); width: 51.6129%;\\"></div></div></li><li class=\\"as-category-widget__category\\"><p class=\\"as-category-widget__info as-body\\"></p><div class=\\"as-category-widget__title\\">Health</div>   400<p></p><div class=\\"as-category-widget__bar\\"><div class=\\"as-category-widget__bar-value\\" style=\\"background-color: var(--category-bar--color); width: 25.8065%;\\"></div></div></li><li class=\\"as-category-widget__category\\"><p class=\\"as-category-widget__info as-body\\"></p><div class=\\"as-category-widget__title\\">Shopping mall</div>   250<p></p><div class=\\"as-category-widget__bar\\"><div class=\\"as-category-widget__bar-value\\" style=\\"background-color: var(--category-bar--color); width: 16.129%;\\"></div></div></li><li class=\\"as-category-widget__category\\"><p class=\\"as-category-widget__info as-body\\"></p><div class=\\"as-category-widget__title\\">Transportation</div> 1.6K<p></p><div class=\\"as-category-widget__bar\\"><div class=\\"as-category-widget__bar-value\\" style=\\"background-color: var(--category-bar--color); width: 100%;\\"></div></div></li><li class=\\"as-category-widget__category\\"><p class=\\"as-category-widget__info as-body\\"></p><div class=\\"as-category-widget__title\\">Leisure</div> 1.2K<p></p><div class=\\"as-category-widget__bar\\"><div class=\\"as-category-widget__bar-value\\" style=\\"background-color: var(--category-bar--color); width: 77.4194%;\\"></div></div></li></ul><footer class=\\"as-category-widget__footer\\"><div class=\\"as-category-widget__count as-body\\">All selected</div><button class=\\"as-btn as-btn--primary as-btn--s as-category-widget__clear\\" disabled=\\"\\">Clear selection</button></footer></as-category-widget>"`;
